python/tests/test_logical_record.py

```diff
@@ -302,6 +282,20 @@ def test_repcode_invalid_in_objects(tmpdir, merge_files_oneLR):
             f.load()
     assert "unknown representation code" in str(excinfo.value)
 
+@pytest.mark.future_warning_repcode_different_no_value
+def test_repcode_different_no_value(tmpdir, merge_files_oneLR):
+    path = os.path.join(str(tmpdir), 'different-repcode-no-value.dlis')
+    content = [
+        'data/chap3/start.dlis.part',
+        'data/chap3/template/default.dlis.part',
+        'data/chap3/object/object.dlis.part',
+        'data/chap3/objattr/csingle-novalue.dlis.part'
+    ]
+    merge_files_oneLR(path, content)
+
+    with dlisio.load(path) as (f, *_):
+        obj = f.object('VERY_MUCH_TESTY_SET', 'OBJECT', 1, 1)
+        assert obj.attic['DEFAULT_ATTRIBUTE']
```

Comment:
This check is not enough. It just checks that DEFAULT_ATTRIBUTE value is not empty.
But we do not check what this value is. Is it still some old floats, or is it now new default csingles, as we want them? We still do not prove that monostate solution worked.

So please check that it equals to complex default values, as per previous comment.